web/js/containers/timeline/timeline.js

```diff
@@ -1284,7 +1284,9 @@ function mapStateToProps(state) {
     hasSubdailyLayers = hasSubDaily(layers[compare.activeString]);
     endTime = layersLastDateTime(layers[activeString], config);
   }
-  endTime = appNow;
+  /** endTime = appNow; This was originally here, wondering why the
+  above logic is there when it gets set here **/
+  endTime = state.config.todaysDate;
```

Comment:
I attempted to make changes, but I still don't think this is behaving exactly as we want it to. An example to try would be to put in the current date using the `now` param (i.e. ?now=2020-03-12T0). The date picker gets set to the correct time which should be yesterday's date. However, advancing to the next date using the right arrow, the right arrow is still in an enabled state.  Clicking it again will disable it and change the param to be the date the page was loaded on which is being captured as `todaysDate`. It's very confusing as we have this other notion of `appNow` which is setting the `endTime` which controls the ending date of the timeline (see Timeline.js). It looks like there is some logic in there that's being completely disregarded as well (lines 1269-1275). I'm somewhat stumped as to how to resolve this effectively. Hoping for another set of eyes on this.

Reply:
I think you found the inherit issue here. You will need to import `testNow` from the `date` state imports above and implement my recommended fix.